src/routes/user.js

```diff
@@ -287,6 +288,183 @@ router.delete(
   })
 );
 
+/**
+ * Begin Flavor Notes
+ */
+
+/**
+ * GET User Flavor Notes
+ * @param userId int
+ */
+router.get(
+  '/:userId(\\d+)/notes',
+  authenticate(),
+  ensurePermission('user', 'read'),
+  [param('userId').isNumeric().isInt({ min: 1 }).toInt()],
+  handleValidationErrors(),
+  handleFindAll(UserFlavorNote, (req) => {
+    const { userId } = req.params;
+
+    log.info(`request flavor notes for user ${userId}`);
+    return {
+      where: {
+        userId
+      },
+      include: [
+        {
+          model: Flavor,
+          required: true,
+          include: [
+            {
+              model: Vendor,
+              required: true
+            }
+          ]
+        },
+        {
+          model: UserProfile,
+          required: true
+        }
+      ]
+    };
+  })
+);
+
+/**
+ * GET A User Flavor Note
+ * @param userId int
+ * @param flavorId int
+ */
+router.get(
+  '/:userId(\\d+)/note/:flavorId(\\d+)',
+  authenticate(),
+  ensurePermission('user', 'read'),
+  [
+    param('userId').isNumeric().isInt({ min: 1 }).toInt(),
+    param('flavorId').isNumeric().isInt({ min: 1 }).toInt()
+  ],
+  handleValidationErrors(),
+  handleFindAll(UserFlavorNote, (req) => {
+    const { userId, flavorId } = req.params;
+
+    log.info(`request flavor note flavor id ${flavorId} for user ${userId}`);
+    return {
+      where: {
+        userId,
+        flavorId
+      },
+      include: [
+        {
+          model: Flavor,
+          required: true,
+          include: [
+            {
+              model: Vendor,
+              required: true
+            }
+          ]
+        },
+        {
+          model: UserProfile,
+          required: true
+        }
+      ]
+    };
+  })
+);
+
+/**
+ * POST Add Flavor Note
+ * @param userId int - User ID
+ */
+router.post(
+  '/:userId(\\d+)/note',
+  authenticate(),
+  ensurePermission('user', 'create'),
```

Comment:
The subject of these ensurePermission should be a newly created `flavorNote` or similar

Reply:
Added the migrations for the `flavorNote` subject and updated these routes